From 4f1dc9c5ab66b07be53c1219a703bcf74c828685 Mon, 10 Oct 2016 15:54:00 +0200
From: Thomas Perl <thomas.perl@jollamobile.com>
Date: Mon, 28 Oct 2013 11:39:52 +0100
Subject: [PATCH] [refresh] Don't resolve URL in zypp_is_changeable_media


This avoids a network roundtrip and fixes a crash related to an uncaught
zypp::PluginScriptException (the alternative would be to handle the
PluginScriptException in zypp_is_changeable_media(), but as CD/DVDs
are not our top priority for distributing nemo, just handle it quickly.

diff --git a/backends/zypp/pk-backend-zypp.cpp b/backends/zypp/pk-backend-zypp.cpp
index fdaef72..a23690e 100644
--- a/backends/zypp/pk-backend-zypp.cpp
+++ b/backends/zypp/pk-backend-zypp.cpp
@@ -818,17 +818,11 @@
 gboolean
 zypp_is_changeable_media (const Url &url)
 {
-	gboolean is_cd = false;
-	try {
-		media::MediaManager mm;
-		media::MediaAccessId id = mm.open (url);
-		is_cd = mm.isChangeable (id);
-		mm.close (id);
-	} catch (const media::MediaException &e) {
-		// TODO: Do anything about this?
-	}
-
-	return is_cd;
+	// Copied from MediaManager::isChangeable() in libzypp's MediaManager.cpp
+	// This avoids a network roundtrip just to check if it's a CD/DVD. We do
+	// not have nemo distributed on optical media, and most of the devices that
+	// run nemo probably won't have an optical drive..
+	return (url.getScheme() == "cd" || url.getScheme() == "dvd");
 }
 
 namespace {
